### PR TITLE
Record check timings

### PR DESCRIPTION
**Pull request by kmclarnon**
First pass at this, needs to be tested but should hopefully be safe to roll out as it's currently opt in via an env var.

@stevegutz @Xcelled

### CODE REVIEW

**Comment by stevegutz** [comment]
Cool! Can we wait to merge til after I upgrade us in https://github.com/HubSpot/error-prone/pull/26 ?

I plan to merge that on Monday

**Comment by kmclarnon** [comment]
Works for me 👍

**Comment by stevegutz** [comment]
🚢  thanks so much for adding these!

File: check_api/src/main/java/com/google/errorprone/HubSpotUtils.java [hunk 1]
```diff
@@ -132,29 +144,51 @@ private static String toInitErrorMessage(Throwable e) {
...
+  }
+
+  private static void flushTimings() {
+    getTimingsOutput().ifPresent(p -> flush(computeFinalTimings(), p));
+  }
+
+  private static Map<String, Long> computeFinalTimings() {
+    TreeMap<String, Long> res = new TreeMap<>(PREVIOUS_TIMINGS);
+    TIMINGS.forEach((k, v) -> {
+      if (res.containsKey(k)) {
```
Comment (stevegutz) [inline check_api/src/main/java/com/google/errorprone/HubSpotUtils.java:157]: Could this be `compute`?

Comment (kmclarnon) [reply check_api/src/main/java/com/google/errorprone/HubSpotUtils.java:172]: Ah yea good call 👍

**Pull request merged**